torch/autograd/graph.py

```diff
@@ -203,6 +204,21 @@ def increment_version(tensor):
     torch._C._increment_version(tensor)
 
 
+def forbid_in_autograd(tensor):
+    """Replaces the current tensor's grad_fn with an Error node.
```

Comment:
While I like the factuality and it makes it clear to me what this does, I think we need a bit more sugar coating for end users. See below.

Also maybe have a small note that this is an advanced API that we don't expect most users to use and we expect that detach() and no_grad() should be used by most users to locally disable autograd as discussed in https://pytorch.org/docs/stable/notes/autograd.html#locally-disabling-gradient-computation